extensions/blocks/slideshow/slideshow.php

```diff
@@ -24,5 +24,161 @@
  */
 function jetpack_slideshow_block_load_assets( $attr, $content ) {
 	Jetpack_Gutenberg::load_assets_as_required( 'slideshow' );
+	if ( class_exists( 'Jetpack_AMP_Support' ) && Jetpack_AMP_Support::is_amp_request() ) {
+		return jetpack_slideshow_block_render_amp( $attr );
+	}
 	return $content;
 }
+
+/**
+ * Render slideshow block for AMP
+ *
+ * @param array $attr Array containing the slideshow block attributes.
+ *
+ * @return string
+ */
+function jetpack_slideshow_block_render_amp( $attr ) {
+	static $wp_block_jetpack_slideshow_id = 0;
+	$wp_block_jetpack_slideshow_id++;
+
+	$ids      = empty( $attr['ids'] ) ? array() : $attr['ids'];
+	$autoplay = empty( $attr['autoplay'] ) ? false : $attr['autoplay'];
+	$classes  = array(
+		'wp-block-jetpack-slideshow',
+		'wp-amp-block',
+		'align' . isset( $attr['align'] ) ? $attr['align'] : 'center',
+		$autoplay ? 'wp-block-jetpack-slideshow__autoplay' : null,
+		$autoplay ? 'wp-block-jetpack-slideshow__autoplay-playing' : null,
+	);
+
+	return sprintf(
+		'<div class="%s" id="wp-block-jetpack-slideshow__%s"><div class="wp-block-jetpack-slideshow_container swiper-container">%s%s%s</div></div>',
+		esc_attr( implode( $classes, ' ' ) ),
+		esc_attr( $wp_block_jetpack_slideshow_id ),
+		jetpack_slideshow_block_amp_carousel( $attr, $wp_block_jetpack_slideshow_id ),
+		$autoplay ? jetpack_slideshow_block_autoplay_ui( $wp_block_jetpack_slideshow_id ) : '',
+		jetpack_slideshow_block_bullets( $ids, $wp_block_jetpack_slideshow_id )
+	);
+}
+
+/**
+ * Generate amp-carousel markup
+ *
+ * @param array $attr Array of block attributes.
+ * @param int   $block_ordinal The ordinal number of the block, used in unique ID.
+ *
+ * @return string amp-carousel markup.
+ */
+function jetpack_slideshow_block_amp_carousel( $attr, $block_ordinal ) {
+	$ids         = empty( $attr['ids'] ) ? array() : $attr['ids'];
+	$first_image = wp_get_attachment_metadata( $ids[0] );
+	$delay       = empty( $attr['delay'] ) ? 3 : intval( $attr['delay'] );
+	$autoplay    = empty( $attr['autoplay'] ) ? false : $attr['autoplay'];
+	return sprintf(
+		'<amp-carousel width="%s" height="%s" layout="responsive" type="slides" data-next-button-aria-label="%s" data-prev-button-aria-label="%s" controls loop %s id="wp-block-jetpack-slideshow__amp-carousel__%s" on="slideChange:wp-block-jetpack-slideshow__amp-pagination__%s.toggle(index=event.index, value=true)">%s</amp-carousel>',
+		esc_attr( $first_image['width'] ),
```

Comment:
While that will work in most cases, we will run into issues on sites with limited resources where folks use large images. In those cases, the image process sometimes fails on upload, and  you end up with missing metadata for your image. I would consequently recommend checking if width and height exist before to try to use them.

Reply:
Good call: https://github.com/Automattic/jetpack/pull/13009/commits/4cd61904a7c6139a02662a3b7d2df0eb98af6553